src/Plugin/Visualisation/Style/AxisChart.php

```diff
@@ -878,7 +878,7 @@ protected function chartBuildSettings(array $records) {
     $tick_values_field = $this->config('axis', 'x', 'tick', 'values', 'field');
 
     if ($settings['axis']['x']['type'] === '' && !empty($record) && property_exists($record, $tick_values_field)) {
-      is_numeric($record->{$tick_values_field}) ?
+      is_numeric($record->{$tick_values_field} && in_array($record->{$tick_values_field}, $settings['axis']['x']['tick']['values']['custom'])) ?
```

Comment:
Looking better @WilsonWu2100 however it looks like there is no guarantee that `$settings['axis']['x']['tick']['values']['custom']` will always be populated tho

maybe
```
$ticks_custom = $settings['axis']['x']['tick']['values']['custom'] ?? [];
is_numeric($record->{$tick_values_field} && in_array($record->{$tick_values_field}, $ticks_custom) ?
```

Also, please make sure this is tested with different data sets and settings

Reply:
@jez500 Code has been updated and tested